pkg/test/expectations/expectations.go

```diff
@@ -83,38 +83,26 @@ func ExpectNotScheduled(ctx context.Context, c client.Client, pod *v1.Pod) *v1.P
 
 func ExpectApplied(ctx context.Context, c client.Client, objects ...client.Object) {
 	for _, object := range objects {
-		if object.GetResourceVersion() == "" {
-			Expect(c.Create(ctx, object)).To(Succeed())
+		current := object.DeepCopyObject().(client.Object)
+		statuscopy := object.DeepCopyObject().(client.Object) // Snapshot the status, since create/update may override
+		deletecopy := object.DeepCopyObject().(client.Object) // Snapshot the status, since create/update may override
+		// Create or Update
+		if err := c.Get(ctx, client.ObjectKeyFromObject(current), current); err != nil {
+			if errors.IsNotFound(err) {
+				Expect(c.Create(ctx, object)).To(Succeed())
+			} else {
+				Expect(err).ToNot(HaveOccurred())
+			}
 		} else {
+			object.SetResourceVersion(current.GetResourceVersion())
 			Expect(c.Update(ctx, object)).To(Succeed())
 		}
-	}
-}
-
-func ExpectStatusUpdated(ctx context.Context, c client.Client, objects ...client.Object) {
-	for _, object := range objects {
-		Expect(c.Status().Update(ctx, object)).To(Succeed())
-	}
-}
-
-func ExpectCreated(ctx context.Context, c client.Client, objects ...client.Object) {
-	for _, object := range objects {
-		Expect(c.Create(ctx, object)).To(Succeed())
-	}
-}
-
-func ExpectCreatedWithStatus(ctx context.Context, c client.Client, objects ...client.Object) {
-	for _, object := range objects {
-		updatecopy := object.DeepCopyObject().(client.Object)
-		deletecopy := object.DeepCopyObject().(client.Object)
-		ExpectApplied(ctx, c, object)
-		// some objects (e.g. PDB) require that the resource version match prior to an update
-		Expect(c.Get(ctx, client.ObjectKeyFromObject(object), object)).To(Succeed())
-		updatecopy.SetResourceVersion(object.GetResourceVersion())
-
-		Expect(c.Status().Update(ctx, updatecopy)).To(Succeed())
+		// Update status
+		statuscopy.SetResourceVersion(object.GetResourceVersion())
+		Expect(c.Status().Update(ctx, statuscopy)).To(Or(Succeed(), MatchError("the server could not find the requested resource"))) // Some objects do not have a status
+		// Delete if timestamp set
 		if deletecopy.GetDeletionTimestamp() != nil {
-			Expect(c.Delete(ctx, deletecopy, &client.DeleteOptions{GracePeriodSeconds: ptr.Int64(int64(time.Until(deletecopy.GetDeletionTimestamp().Time).Seconds()))})).ToNot(HaveOccurred())
+			Expect(c.Delete(ctx, deletecopy)).To(Succeed())
```

Comment:
Can we add a grace period of zero seconds here?  I've added it a few other places to reduce test flakes where objects sometimes hang around.